### PR TITLE
Improve HF qwen3_omni: preserve audio_sample_rate in kwargs restructuring

### PR DESCRIPTION
**Pull request by jeremyteboul**
The Qwen3OmniMoeProcessor was losing the audio_sample_rate parameter during kwargs restructuring for transformers < 4.58.0. When mm_kwargs were reorganized into audio_kwargs and text_kwargs dictionaries, the audio_sample_rate (passed at the top level) was not being moved into audio_kwargs where the HuggingFace WhisperFeatureExtractor expects it.

This caused audio processing to fail with:
  Failed to apply Qwen3OmniMoeProcessor on data={'audio': [array(...)]}
  with kwargs={'audio_sample_rate': 16000, 'audio_kwargs': {}, ...}

Changes:
- Extract audio_sample_rate before kwargs restructuring
- Place it into audio_kwargs after creating nested dictionaries
- Add comprehensive unit tests for various sample rates

Tests:
pytest tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation -v

Test coverage:
- test_audio_sample_rate_preserved_in_audio_kwargs: Core fix validation
- test_audio_sample_rate_absent_when_not_provided: Edge case handling
- test_various_audio_sample_rates_preserved: Parameterized test for 8kHz, 16kHz, 22kHz, 24kHz, 44kHz, and 48kHz sample rates

All 8 tests passing:
  tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_audio_sample_rate_preserved_in_audio_kwargs PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_audio_sample_rate_absent_when_not_provided PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_various_audio_sample_rates_preserved[8000] PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_various_audio_sample_rates_preserved[16000] PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_various_audio_sample_rates_preserved[22050] PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_various_audio_sample_rates_preserved[24000] PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_various_audio_sample_rates_preserved[44100] PASSED tests/multimodal/test_processing.py::TestQwen3OmniAudioSampleRatePreservation::test_various_audio_sample_rates_preserved[48000] PASSED ========================= 8 passed in 0.15s =========================
  
  Signed-off-by: Jeremy Teboul <jeremyte@meta.com>

### CODE REVIEW

File: tests/multimodal/test_processing.py [hunk 1]
```diff
@@ -1027,21 +1027,166 @@ def test_hf_processor_init_kwargs(
...
+        model_id=model_id,
+# Test Qwen3 Omni audio_sample_rate preservation
+class TestQwen3OmniAudioSampleRatePreservation:
+    """Test that audio_sample_rate is preserved during kwargs restructuring.
+
+    These tests validate the fix for the audio_sample_rate bug in Qwen3 Omni
+    where the parameter was lost during kwargs restructuring. The tests don't
+    require importing the actual model classes - they just test the kwargs
+    manipulation logic.
+    """
```
Comment (Isotr0py) [inline tests/multimodal/test_processing.py]: Since this is a special case for Qwen3-omni, let's create test at `tests/models/multimodal/processing/test_qwen3_omni.py` instead of `test_common.py`.

File: tests/multimodal/test_processing.py [hunk 2]
```diff
@@ -1027,21 +1027,166 @@ def test_hf_processor_init_kwargs(
...
+
+    def test_audio_sample_rate_preserved_in_audio_kwargs(self) -> None:
+        """
+        Test that audio_sample_rate is moved from top-level mm_kwargs
+        into audio_kwargs during kwargs restructuring.
+
+        This is the core fix: when transformers < 4.58.0, the code
+        restructures kwargs into audio_kwargs and text_kwargs, and
+        audio_sample_rate must be preserved in audio_kwargs.
+        """
```
Comment (Isotr0py) [inline tests/multimodal/test_processing.py]: Seems that this test is not an e2e test like others. Can you add an e2e ones? You can refer to `tests/models/multimodal/processing/test_qwen2_vl.py`.

File: tests/multimodal/test_processing.py [hunk 3]
```diff
@@ -1027,23 +1027,165 @@ def test_hf_processor_init_kwargs(
...
+                "truncation": tok_kwargs_copy.pop("truncation", False)
+            }
+            if audio_sample_rate_val is not None:
+                mm_kwargs_copy["audio_kwargs"]["audio_sample_rate"] = (
+                    audio_sample_rate_val
+                )
+
+        # Assert: Verify the specific sample rate is preserved
+        assert mm_kwargs_copy["audio_kwargs"]["audio_sample_rate"] == sample_rate
+
```
[the remark this answers is not in the document]
Comment (jeremyteboul) [reply tests/multimodal/test_processing.py]: command have been addressed

File: tests/multimodal/test_processing.py (hunk 1 above)
Comment (jeremyteboul) [reply tests/multimodal/test_processing.py]: done I moved the test in the new file

File: tests/multimodal/test_processing.py (hunk 2 above)
Comment (jeremyteboul) [reply tests/multimodal/test_processing.py]: Added also a e2e test ; thanks for the suggestion; let me know if we miss anything else

File: tests/multimodal/test_processing.py [hunk 4]
```diff
@@ -1027,23 +1027,165 @@ def test_hf_processor_init_kwargs(
...
-    )
-
-    processor = ctx.get_hf_processor(
-        DummyProcessor,  # type: ignore[arg-type]
-        **inference_kwargs,
+    processor = _get_dummy_processor(
+        model_id=model_id,
+        config_kwargs=config_kwargs,
+        inference_kwargs=inference_kwargs,
     )
```
[the remark this answers is not in the document]
Comment (jeremyteboul) [reply tests/multimodal/test_processing.py]: This is fixed

File: tests/multimodal/test_processing.py [hunk 5]
```diff
@@ -1039,9 +1039,8 @@ def test_hf_processor_init_kwargs(
         DummyProcessor,  # type: ignore[arg-type]
         **inference_kwargs,
     )
-
-    for k, v in expected_kwargs.items():
-        assert getattr(processor, k) == v
```
Comment (Isotr0py) [inline tests/multimodal/test_processing.py:1026]: Is this change necessary?

Comment (jeremyteboul) [reply tests/multimodal/test_processing.py:1026]: I believe so  now we only check known init parameters:
It makes the test more explicit about what it's checking (init params only)
It's robust to kwargs restructuring and  It aligns with proper separation of concerns (init vs call params)

File: tests/models/multimodal/processing/test_qwen3_omni.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+from typing import Any
+
+import pytest
+
+
+class TestQwen3OmniAudioSampleRatePreservation:
```
Comment (Isotr0py) [inline tests/models/multimodal/processing/test_qwen3_omni.py:111]: I think we have better validate that vLLM's processor can also generate expected outputs with different `"audio_sample_rate"`.

So the test will be something like this:
```python3
    # Build the image str / prompt based on the number of images we pass
    prompt = "<|audio_start|><|audio_pad|><|audio_end|>"
    mm_data = ...
    
    hf_processor_mm_kwargs = {"audio_sampling_rate": ... }
    processed_inputs = processor.apply(prompt, mm_data, hf_processor_mm_kwargs)

    # Ensure we have the right number of placeholders per num_crops size
    hf_processor = processor.info.get_hf_processor(**hf_processor_mm_kwargs)
    audio_token_id = tokenizer.convert_tokens_to_ids(hf_processor.audio_token)
    aud_tok_count = processed_inputs["prompt_token_ids"].count(audio_token_id)
    assert aud_tok_count == expected_aud_tok_count
```

Comment (jeremyteboul) [reply tests/models/multimodal/processing/test_qwen3_omni.py:111]: @Isotr0py , sorry for late reply, here is the missing test

**Review by Isotr0py — APPROVED** [review]
